Zend/tests/pipe_operator/ast.phpt

```diff
@@ -0,0 +1,25 @@
+--TEST--
+Test that a pipe operator displays as a pipe operator when outputting syntax.
+--FILE--
+<?php
+
+function _test(int $a): int {
+    return $a + 1;
+}
+
+try {
+    assert((5 |> '_test') == 99);
+} catch (AssertionError $e) {
+    print $e->getMessage() . PHP_EOL;
+}
+
+try {
+    assert((5 |> _test(...)) == 99);
+} catch (AssertionError $e) {
+    print $e->getMessage() . PHP_EOL;
+}
+
+?>
+--EXPECTF--
+assert(5 |> '_test' == 99)
+assert(5 |> _test(...) == 99)
```

Comment:
The RFC claims that the pipe operator has a low precedence. Are parentheses missing around the pipe operator here in the output or is `==` lower than the pipe operator? In that case, also add a test case where the parentheses are actually necessary.

Reply:
I see that you added a comparison operator test, but that's now what I was looking for: I was commenting on the AST-printing test, thus I expected it to be adjusted / fixed.

Reply:
I don't understand what you're asking for.  Can you clarify and/or suggest code for it?

Reply:
Oh!  I see.  So it's not an issue with the test per se, it's that the AST printer is omitting the parens when it shouldn't.  (And fixing that would include updating the test.)  Hm.  

In that case I'm not sure why the code in https://github.com/php/php-src/pull/17118/files#diff-129a10943d4977ae47c23ce48aa2b29f499aa1f04fd94d6aa48b6b0a4bd78175R2509 is skipping the parens.  It should just be taking "whatever the render of the left side is", then pipe, then "whatever the render of the right side", right?

Reply:
> It should just be taking "whatever the render of the left side is", then pipe, then "whatever the render of the right side", right?

The AST does not contain parentheses, because the grouping is made clear by the tree structure (the children would look differently if the parentheses were placed differently).

My untested understanding is that can use the `BINARY_OP()` helper macro to pretty-print the AST with the correct parentheses, see `ZEND_AST_AND` et al for an example.